include/data/constructs/rfile/bcfile/BlockCompressedFile.h

```diff
@@ -171,8 +171,18 @@ class BlockCompressedFile : public cclient::data::streams::StreamInterface
         }
         delete[] magicVerify;
         // get the index meta
-        in_stream->seek (fileLength - magic_size - VERSION_SIZE - 8);
-        offsetIndexMeta = in_stream->readLong ();
+        if (version.getMajor() == 1)
+        {
+          in_stream->seek (fileLength - magic_size - VERSION_SIZE - 8);
+          offsetIndexMeta = in_stream->readLong ();
+        }
+        else{
```

Comment:
Is this the main change to the persisted data?

Reply:
There was one other place. Keep in mind that I already supported multi level indices and prefix encoding, so this was a very minor update. Will add some tests for different versions.